Loss function for multilabel_soft_margin_loss in paddle frontend

Indraneel99 opened the pull request:
Close #20797

Indraneel99 commented:
Hi team,

Some checks are failing (2 in 128), does this mean I am missing edge cases? Please let me know where can i find crash log ( I am unable to find it 😟). 

I will update the formatting. Thanks

hirwa-nshuti commented on ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py:
```diff
@@ -390,3 +390,48 @@ def test_paddle_margin_ranking_loss(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        num_arrays=2,
+        exclude_min=True,
```
I believe that we can add this if we set the bounds of both `max` and `min` values of arrays to be generated

hirwa-nshuti reviewed (changes requested):
Hey @Indraneel99 

I tried to run your tests locally and I'm getting the following errors
![image](https://github.com/unifyai/ivy/assets/67042527/f1112ed3-ea71-4251-82e9-e1b0f8512e63)

Would you check it out and the comments I left below🙂

hirwa-nshuti commented on ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py:
```diff
@@ -390,3 +390,50 @@ def test_paddle_margin_ranking_loss(
+    x_dtype, x = dtype_and_x
+    weight_dtype, weight = dtype_and_weight
+    helpers.test_frontend_function(
+        input_dtypes=[
```
This parameter is supposed to be a list, not a nested list, you can concatenate both `x_dtype` and `weight_dtype` lists

hirwa-nshuti commented on ivy/functional/frontends/paddle/nn/functional/loss.py:
```diff
@@ -230,3 +230,21 @@ def margin_ranking_loss(input, other, label, margin=0.0, reduction="mean", name=
+
+@with_supported_dtypes({"2.5.1 and below": ("float32","float64")},"paddle")
+@to_ivy_arrays_and_back
+def multilabel_soft_margin_loss(input, label, weight=None, reduction='mean', name=None):
```
Function name is written incorrectly would you rename it to `multi_label_soft_margin_loss`

hirwa-nshuti commented on ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py:
```diff
@@ -390,3 +390,50 @@ def test_paddle_margin_ranking_loss(
     )
+
+@handle_frontend_test(
+    fn_tree="paddle.nn.functional.multilabel_soft_margin_loss",
```
The function should also be `multi_label_soft_margin_loss`

hirwa-nshuti reviewed (changes requested):
Hey I added a few comments to help you run tests successfully for you to debug the failing tests

Indraneel99 commented:
Hi @hirwa-nshuti , Thanks for the review and comments. I have changed the function name to multi_label_soft_margin_loss and updated the code. All Tests are passing locally, but it seems some checks are failing in GitHub. please help me with the failing checks. Thanks

hirwa-nshuti reviewed (changes requested):
Would you also try tp fix the merge conflicts so I can be able to run the tests locally😊
Thanks

Indraneel99 commented:
> Would you also try tp fix the merge conflicts so I can be able to run the tests locally😊 Thanks

I have resolved the conflicts. Please look into it. Thanks ☺️

hirwa-nshuti commented on ivy/functional/frontends/paddle/nn/functional/loss.py:
```diff
@@ -315,3 +315,54 @@ def nll_loss(
+  
+@with_supported_dtypes({"2.5.1 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def nll_loss(
```
This should be removed as it was redefined from above

hirwa-nshuti reviewed (changes requested):
Hey @Indraneel99 

Good job so far 😊
There are assertion errors in backends would you look into this🙂
Thanks

Indraneel99 commented:
Hey @hirwa-nshuti, Thanks for reviewing (sorry for the earlier mistakes 😢 ). I have updated the function and now all tests are passing locally. Please check it. Thanks 😄.

hirwa-nshuti reviewed (changes requested):
Hey @Indraneel99 

Sorry for getting back to you this late 🙂
Would you resolve these merge conflicts so that the CI check should run
Thanks!

Indraneel99 commented:
Hey @hirwa-nshuti . I have resolved the conflicts. Please check the code. Thanks.

hirwa-nshuti reviewed (approved):
Hey @Indraneel99 

All looks great to me 😊
Thanks for contributing

Indraneel99 commented:
Hey @hirwa-nshuti . Thanks for all the help 🙂.
